src/submission-handlers/base-handlers.ts

```diff
@@ -65,7 +65,7 @@ export const ObsSubmissionHandler: SubmissionHandler = {
       if (rendering == 'radio' || rendering == 'content-switcher') {
         getConcept(field.questionOptions.concept, 'custom:(uuid,display,datatype:(uuid,display,name))').then(
           (result) => {
-            if (result.datatype.name == 'Boolean') {
+            if (result?.datatype.name == 'Boolean') {
```

Comment:
```suggestion
            if (result?.datatype.name === 'Boolean') {
```

=== is preferred because == will do type coercion.